build.ps1

```diff
@@ -157,10 +157,15 @@ Invoke-BuildStep $VSMessage {
 -ev +BuildErrors
 
 Invoke-BuildStep 'Creating the EndToEnd test package' {
-        param($Configuration)
-        $EndToEndScript = Join-Path $PSScriptRoot scripts\cibuild\CreateEndToEndTestPackage.ps1 -Resolve
-        $OutDir = Join-Path $Artifacts VS15
-        & $EndToEndScript -c $Configuration -out $OutDir
+        $msbuildArgs = "test\TestUtilities\CreateEndToEndTestPackage\CreateEndToEndTestPackage.proj", "/p:Configuration=$Configuration", "/restore:false", "/property:BuildProjectReferences=false"
```

Comment:
I think that MSBuild (at least outside of MSFT employees) is significantly less well known, and much harder to find help online than powershell.  PowerShell also has PowerShell ISE that allows step by step debugging.

So, I think that converting powershell scripts into MSBuild is a step backwards.

Reply:
In some cases I agree.  If this Powershell script ran locally and was used by us, Powershell could be the right answer.  I am not advocating that we take every Powershell script we have and replace it with MSBuild logic.

However, in this case the script is doing the following:

1. Assumes `bootstrap.proj` has been restored via `MSBuild /t:restore` or `dotnet restore`
2. Parses `bootstrap.proj` as XML so it knows the version of a package and forms a path to the NuGet package that was restored by `bootstrap.proj`.
3. Gets the output after _building_ `API.Test.csproj` and copies them to a specific location
4. Gets the output after _building_ `GenerateTestPackages.csproj` and copies them to a specific location
5. Copies files from the repository to a specific location

Most of that is relying on `MSBuild` to have restored packages and built projects.  Replacing the logic with an MSBuild project itself with standard `<ProjectReference />` items, a set of items, and a call to a `<Robocopy />` task doesn't seem like that big of a shift.

Also, this script runs on a remote machine most of the time and debugging what Powershell is doing remotely isn't great in my opinion.  The MSBuild binary logger is great for capturing what took place on a different machine.